parser/parser.go

```diff
@@ -126,48 +126,48 @@ func (p *parser) expect(kind Kind, values ...string) {
 // parse functions
 
 func (p *parser) parseExpression(precedence int) Node {
-	if precedence == 0 {
-		if p.current.Is(Operator, "let") {
-			return p.parseVariableDeclaration()
-		}
+	if precedence == 0 && p.current.Is(Operator, "let") {
+		return p.parseVariableDeclaration()
 	}
 
 	nodeLeft := p.parsePrimary()
 
 	prevOperator := ""
-	opToken := p.current
-	for opToken.Is(Operator) && p.err == nil {
-		negate := false
+	for opToken := p.current; opToken.Is(Operator) && p.err == nil; opToken = p.current {
```

Comment:
I think it's better to keep the old formatting, as we iterating not only on `opToken`, but also updating `prevOperator` on each cycle.

Reply:
Reverted to old format as per your feedback.